journal/src/main/java/io/camunda/zeebe/journal/file/JournalSegments.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.journal.file;
+
+import io.camunda.zeebe.journal.JournalException;
+import io.camunda.zeebe.journal.file.record.CorruptedLogException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.SortedMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JournalSegments {
+
+  private static final int FIRST_SEGMENT_ID = 1;
+  private static final int INITIAL_INDEX = 1;
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final JournalMetrics journalMetrics;
+  private final NavigableMap<Long, JournalSegment> segments = new ConcurrentSkipListMap<>();
+  private volatile JournalSegment currentSegment;
+
+  private final JournalIndex journalIndex;
+  private final int maxSegmentSize;
+
+  private final File directory;
+
+  private final SegmentFileManager segmentLoader;
+
+  private final long lastWrittenIndex;
+
+  private final String name;
+
+  public JournalSegments(
+      final JournalMetrics journalMetrics,
+      final JournalIndex journalIndex,
+      final int maxSegmentSize,
+      final File directory,
+      final long lastWrittenIndex,
+      final String name) {
+    this.journalMetrics = journalMetrics;
+    this.journalIndex = journalIndex;
+    this.maxSegmentSize = maxSegmentSize;
+    this.directory = directory;
+    this.lastWrittenIndex = lastWrittenIndex;
+    this.name = name;
+    segmentLoader = new SegmentFileManager(lastWrittenIndex, this.journalIndex);
```

Comment:
:wrench: I would rather inject the `SegmentFileManager` instead of constructing it. It might also make it easier to test users of this class (e.g. `SegmentedJournal`) if we can provide a different implementation (though I'd have to look if that's currently the case, I don't have all the tests OTOH now).